From b04670bccd0ec722a371ff2bfd3b77cee6d769fb Mon Sep 17 00:00:00 2001
From: Ghouse Adoni <ghouse.adoni.mohammed@intel.com>
Date: Fri, 23 Sep 2016 09:41:35 -0700
Subject: [PATCH] kernel: Adding support for HW flip using AtomISP

Enabling driver to program ISP for HW 180 flip when
it is set by the application.

Tracked-on:https://jira01.devtools.intel.com/browse/CR-156
Signed-off-by: Jian Xu Zheng <jian.xu.zheng@intel.com>
Signed-off-by: Eddie Luo <eddie.luo@intel.com>
Signed-off-by: Ghouse Adoni <ghouse.adoni.mohammed@intel.com>
---
 .../media/pci/atomisp/atomisp_driver/atomisp_ioctl.c   | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/drivers/media/pci/atomisp/atomisp_driver/atomisp_ioctl.c b/drivers/media/pci/atomisp/atomisp_driver/atomisp_ioctl.c
index 8ee6e52..c95df80 100644
--- a/drivers/media/pci/atomisp/atomisp_driver/atomisp_ioctl.c
+++ b/drivers/media/pci/atomisp/atomisp_driver/atomisp_ioctl.c
@@ -61,6 +61,22 @@ static const u32 VERSION = DRIVER_VERSION;
  */
 static struct v4l2_queryctrl ci_v4l2_controls[] = {
 	{
+		.id = V4L2_CID_VFLIP,
+		.type = V4L2_CTRL_TYPE_BOOLEAN,
+		.name = "Vertical flip",
+		.minimum = 0,
+		.maximum = 1,
+		.step = 1,
+		.default_value = 0,
+	}, {
+		.id = V4L2_CID_HFLIP,
+		.type = V4L2_CTRL_TYPE_BOOLEAN,
+		.name = "Horizontal flip",
+		.minimum = 0,
+		.maximum = 1,
+		.step = 1,
+		.default_value = 0,
+	}, {
 		.id = V4L2_CID_AUTO_WHITE_BALANCE,
 		.type = V4L2_CTRL_TYPE_BOOLEAN,
 		.name = "Automatic White Balance",
@@ -2298,6 +2314,8 @@ static int atomisp_s_ctrl(struct file *file, void *fh,
 	switch (control->id) {
 	case V4L2_CID_AUTO_N_PRESET_WHITE_BALANCE:
 	case V4L2_CID_EXPOSURE:
+	case V4L2_CID_HFLIP:
+	case V4L2_CID_VFLIP:
 	case V4L2_CID_EXPOSURE_AUTO:
 	case V4L2_CID_EXPOSURE_AUTO_PRIORITY:
 	case V4L2_CID_SCENE_MODE:
-- 
1.9.1

